src/schema.cpp

```diff
@@ -97,7 +97,7 @@ struct IsNotRemoveProperty {
 struct GetRemovedColumn {
     size_t operator()(SchemaChange sc) const { return sc.visit(*this); }
     size_t operator()(schema_change::RemoveProperty p) const { return p.property->table_column; }
-    template<typename T> size_t operator()(T) const { __builtin_unreachable(); }
+    template<typename T> size_t operator()(T) const { REALM_UNREACHABLE(); }
```

Comment:
This change has an unfortunate impact on the compiled binary size since `REALM_UNREACHABLE()` is not equivalent to `__builtin_unreachable()` (it's `abort()` rather than UB) and clang's optimizer is not smart enough to detect that this is impossible to hit.